docs/docs/lists-conditional-rendering.md

```diff
@@ -0,0 +1,220 @@
+
+```javascript{11}
+function NumberList(props) {
+  const numbers = props.numbers
+  const listItems = numbers.map((item) => <li key={"item-" + item}>{item}</li>);
+  return <ul>{listItems}</ul>
+}
+
+const numbers = [1, 2, 3, 4, 5];
+ReactDOM.render(<NumberList numbers={numbers} />, document.getElementById('root'));
+```
+
+[Try it out on Codepen.](https://codepen.io/ericnakagawa/pen/gwjdzN?editors=0011)
+
+
+### Keys
+
+Keys are important for helping React know if items have changed, are added, or removed.
+
+Use keys that represent the identity of the item. If you don't have a way to uniquely identify each item consider using a string representation of the item. _You will be warned if you don't include keys._
+
+Supply keys to elements inside a `map()` function and not directly to an HTML element. Keys used within groups should be unique to each other. Keys cannot be accessed as a `prop`.
+
+**Example: Incorrect key usage:** _Should not assign a `key` in this manner._
+
+```javascript{2}
+function Number(props) {
+  return <li key={"item-" + props.value}>{props.value}</li>;
+}
+
+function NumberList(props) {
+  const numbers = props.numbers
+  return <ul>{numbers.map((item) => <Number value={item} />}</ul>
+}
+
+const numbers = [1, 2, 3, 4, 5];
+ReactDOM.render(<NumberList numbers={numbers} />, document.getElementById('root'));
+```
+
+**Example: Correct key usage:** _Assigns `key` inside `map()` function._
+
+```javascript{7}
+function Number(props) {
+  return <li>{props.value}</li>;
+}
+
+function NumberList(props) {
+  const numbers = props.numbers
+  return <ul>{numbers.map((item) => <Number key={"item-" + props.value} value={item} />)}</ul>
+}
+
+const numbers = [1, 2, 3, 4, 5];
+ReactDOM.render(<NumberList numbers={numbers} />, document.getElementById('root'));
+```
+
+[Try it out on Codepen.](https://codepen.io/ericnakagawa/pen/Egpdrz?editors=0010)
+
+## Conditional Rendering
+
+You can create distinct components that encapsulate behavior you need. Then, you can render only the specific component you need, depending on the state of your application.
+
+### Element Variables
+
+You can use variables to store elements. This can help you render the specific element you need based on the state of your application.
+
+In the example below, we want to display a login or logout button. In `render()` we check `state.loggedIn` and assign to `button` either `<LoginButton />` or `<LogoutButton />`. We then render the element.
+
+```javascript{23,25,29}
+function LogoutButton(props) {
+  return <button onClick={props.onClick}>Logout</button>
+}
+
+function LoginButton(props) {
+  return <button onClick={props.onClick}>Login</button>
+}
+
+class LoginControl extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {loggedIn: true};
+    this.login = this.login.bind(this);
+    this.logout = this.logout.bind(this);
+  }
+  login() {
```

Comment:
Maybe call them handleLoginClick and handleLogoutClick to match the most common convention